functorch/csrc/DynamicLayer.h

```diff
@@ -11,16 +11,18 @@ namespace at {
 namespace functorch {
 
 struct TORCH_API DynamicLayer {
-  DynamicLayer(DispatchKey key, int64_t layerId): key_(key), layerId_(layerId) {}
+  DynamicLayer(DispatchKey key, int64_t layerId, optional<int64_t> batchSize = nullopt): key_(key), layerId_(layerId), batchSize_(batchSize) {}
 
   DispatchKey key() const { return key_; }
   int64_t layerId() const { return layerId_; }
+  int64_t batchSize() const { return *batchSize_; }
```

Comment:
this should probably `TORCH_INTERNAL_ASSERT(batchSize_.defined())` to avoid future segfaults